### PR TITLE
Static Webapp Azure Frontdoor Integration (enterprise edge) 

### PR DESCRIPTION
**Pull request by StrawnSC**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [X] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Webapp

**Review by panchagnula — APPROVED** [review]
LGTM

**Comment by StrawnSC** [comment]
Putting `do-not-merge` label on again so we can wait to release after an announcement (at @simonaco's request)

**Comment by StrawnSC** [comment]
Should be merged on the evening (in PT) of 11 Jan 2022

File: src/enterprise-edge/azext_enterprise_edge/custom.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        registration = send_raw_request(cmd.cli_ctx, "GET", request_url).json().get("registrationState").lower()
+        if registration != "registered":
+            raise CLIError("Provider Microsoft.CDN is not registered. "
+                           "Please run 'az provider register --wait --namespace Microsoft.CDN'")
+
+    @classmethod
+    def _validate_sku(cls, sku_name):
+        if sku_name.lower() != "standard":
+            raise CLIError("Invalid SKU: '{}'. Staticwebapp must have 'Standard' SKU to use "
+                           "enterprise edge CDN".format(sku_name))
```
Comment (zhoxing-ms) [inline src/enterprise-edge/azext_enterprise_edge/custom.py:72]: May I ask why is this client not defined in Python SDK?

**Comment by zhoxing-ms** [comment]
Could you add some tests for those new commands?

**Comment by StrawnSC** [comment]
> Could you add some tests for those new commands?

@zhoxing-ms we don't have any scenario tests for static webapps (even in the core CLI) because the create command requires a GitHub auth token, which would fail the cred scan

File: src/enterprise-edge/azext_enterprise_edge/custom.py (hunk 1 above)
Comment (StrawnSC) [reply src/enterprise-edge/azext_enterprise_edge/custom.py:72]: These new API features were not added to the SDK in time, so we decided to release an extension in the meantime that calls the ARM API directly. Once the SDK is ready and we want to incorporate this feature into the core CLI, we will change it to use the SDK 
@panchagnula

Comment (zhoxing-ms) [reply src/enterprise-edge/azext_enterprise_edge/custom.py:72]: OK

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**